src/ts_config.h

```diff
@@ -75,4 +75,24 @@
 #define TS_BYTE_STRING_TYPE_SUPPORT CONFIG_THINGSET_BYTE_STRING_TYPE_SUPPORT
 #endif
 
+/*
+ * The ThingSet specification v0.5 introduces a different data layout compared to previous
+ * versions where the data is grouped by entities of the device (like battery, actuator)
+ * instead of the data type (e.g. measurement, configuration). The data type is described by
+ * a single character prefix in the item name.
+ * The new grouping allows to have same item names in different groups, so for unambigous
+ * description of the data the nested structure has to be maintained in statements.
+ *
+ * With nested JSON enabled, requesting names for IDs will also return the entire path (as
+ * a JSON pointer) instead of just the data item name.
+ *
+ * This option is introduced to maintain compatibility with legacy firmware and will be
+ * enabled by default in the future.
+ */
```

Comment:
Maybe the ThingSet specification v0.5 should introduce a mandatory ThingSet data object to define the maximum protocol version a device supports. This could be given a fixed data object ID >=  2048 (0x8000) in the control objects ID space.

Reply:
Let's continue this discussion a separate issue in the specification repository.